specification/redis/resource-manager/Microsoft.Cache/stable/2022-05-01/redis.json

```diff
@@ -0,0 +1,2815 @@
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redis/{name}": {
+      "put": {
+        "tags": [
+          "Redis"
+        ],
+        "operationId": "Redis_Create",
+        "x-ms-examples": {
+          "RedisCacheCreateDefaultVersion": {
+            "$ref": "./examples/RedisCacheCreateDefaultVersion.json"
+          },
+          "RedisCacheCreateLatestVersion": {
+            "$ref": "./examples/RedisCacheCreateLatestVersion.json"
+          },
+          "RedisCacheCreate": {
+            "$ref": "./examples/RedisCacheCreate.json"
+          }
+        },
+        "description": "Create or replace (overwrite/recreate, with potential downtime) an existing Redis cache.",
+        "x-ms-long-running-operation": true,
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the Redis cache."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RedisCreateParameters"
+            },
+            "description": "Parameters supplied to the Create Redis operation."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "The new redis cache was successfully created. Check provisioningState to see detailed status.",
+            "schema": {
+              "$ref": "#/definitions/RedisResource"
+            }
+          },
+          "200": {
+            "description": "The existing redis cache was successfully updated. Check provisioningState to see detailed status.",
+            "schema": {
+              "$ref": "#/definitions/RedisResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "patch": {
+        "tags": [
+          "Redis"
+        ],
+        "operationId": "Redis_Update",
+        "x-ms-long-running-operation": true,
```

Comment:
>ms [](http://example.com/codeflow?start=11&length=2)

converting a sync API to async is considered a cross version breaking change. Why is this being done? Can you achieve the update using a long running POST instead?

Reply:
We've already gotten a signoff for these async-related breaking changes. They were part of a [previous PR](https://github.com/Azure/azure-rest-api-specs/pull/17320), but our team decided to include them in a new API version instead. @chantalolieman can provide more context if needed.